mig-agent/context.go

```diff
@@ -275,7 +275,7 @@ mqdone:
 
 func initChannels(orig_ctx Context) (ctx Context, err error) {
 	ctx = orig_ctx
-	ctx.Channels.Terminate = make(chan string)
+	ctx.Channels.Terminate = make(chan string, 12)
```

Comment:
You're violating the unwritten rule that the buffer size should be a prime number, for no particular reason.